src/avt/DBAtts/MetaData/avtDatabaseMetaData.C

```diff
@@ -6299,6 +6294,19 @@ avtDatabaseMetaData::MeshForVar(const std::string &invar) const
         }
     }
 
+    // If the variable is compound, parse out the variable name.
```

Comment:
Ok, so to understand what was going on...

- Plugin *publishes* variable named `rho(H20)`
- VisIt *mangles* that varaible to `rho_lp_H2O_rp_` and it is this name that VisIt knows this variable by.
- However, plugin itself does some work of its own and tries to use `avtDatabaseMetaData::MeshForVar` but using the string *it* knows the variable as, `rho(H2O)`. It turns out that VisIt's `VarIsCompound()` method looks only for a string of the form `foo(BAR)` and if found, it returns `BAR` as the answer. IMHO, this is the crux of the problem, `VarIsCompound()` assumes that *any* string of this form is a string that VisIt constructed for a *compound* variable name when the truth is, it could have been a variable name from the plugin itself.

So, I don't think moving the `VarIsCompound()` check actually fixes the real problem though it does allow the tested case to pass.

What I think needs to be done in `VarIsCompound` is to a) compute the var part `foo` in `foo(BAR)` and compute the mesh part, `BAR` in `foo(BAR)` and then confirm `foo` is a variable it knows about and `BAR` is a mesh it knows about. If all of that is true, `VarIsCompound()` returns true, otherwise false. 

However, I worry about performance implications because it potentially means we do brute force searching through lists of variables everywhere `VarIsCompound()` is called then. So, maybe don't do the var-part in the above, only confirm the mesh part is indeed a mesh it knows.

Reply:
Yeah, this crossed my mind as well. I didn't see this method being used too extensively, and it looked fairly safe elsewhere. I can pull out the segment in parens and check that to be extra safe.

Reply:
Sure...and then leave the logic in the place it was here instead of moving to end also, ok?